### PR TITLE
Add an end to end QA test

### PR DESCRIPTION
**Pull request by jkakavas**
A single ES cluster is used as an IDP and an SP and the browser is
simulated by the rest client.

### CODE REVIEW

**Comment by jkakavas** [comment]
This only adds an IDP initiated SSO test, SP initiated is incoming.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRequestHandler.java [hunk 1]
```diff
@@ -141,14 +143,20 @@ void validateSignature(Signature signature) {
         }
 
         checkIdpSignature(credential -> {
-            try (RestorableContextClassLoader ignore = new RestorableContextClassLoader(SignatureValidator.class)) {
-                SignatureValidator.validate(signature, credential);
-                logger.debug(() -> new ParameterizedMessage("SAML Signature [{}] matches credentials [{}] [{}]",
-                        signatureText, credential.getEntityId(), credential.getPublicKey()));
-                return true;
+            try {
```
Comment (jkakavas) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRequestHandler.java]: This is rather peculiar, but I can't figure out why the doPrivileged block is necessary _now_ when the cluster is run in our QA project

Comment (tvernum) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRequestHandler.java]: We discussed on Zoom. The RestorableContextClassLoader was moved from security to core in the IDP branch, but it looks like all the tests that cover this are disabled.

**Comment by jkakavas** [comment]
Superceded by #54215

**Pull request closed without merging**